### PR TITLE
[EJBCLIENT-302] Make following timeouts configurable and change default value: initialModuleAvailabilityraise, reconnect.tasks.timeout and version handshake

### PR DESCRIPTION
**Pull request by JiriOndrusek**
Cherry-picked from d203513 + changed default value to 30 + as privileged + 2 more timeouts

Issue:  https://issues.jboss.org/browse/EJBCLIENT-302

No upstream required: as @dmlloyd confiremed: "These timeouts don't make sense in 4.x"

### CODE REVIEW

File: src/main/java/org/jboss/ejb/client/EJBClientContext.java [hunk 1]
```diff
@@ -73,6 +73,12 @@
 
     private static final EJBClientInterceptor.Registration[] NO_INTERCEPTORS = new EJBClientInterceptor.Registration[0];
 
+    /**
+     * Timeout defined through system property "org.jboss.ejb.received-cluster-topology-wait-time", value in property is in seconds.
+     */
+    public static final int RECEIVED_CLUSTER_TOPOLOGY_WAIT_TIMEOUT;
```
Comment (dmlloyd) [inline src/main/java/org/jboss/ejb/client/EJBClientContext.java]: This should not be public as that constitutes API.  If it needs to be shared between packages, perhaps a holder class in the _private package is more appropriate.

**Comment by JiriOndrusek** [comment]
@dmlloyd you are right. I missed this problem There is no need to share these constants between packages, so I fixed it to private (as it should be originally)

**Comment by fl4via** [comment]
AFAICS, the suggested changes have been approved. I don't understand why github still shows the status as Changes Requested.

**Comment by fl4via** [comment]
@JiriOndrusek can you please create a Jira for this issue and add the Jira number to the commit and PR? Thanks a lot!

**Comment by JiriOndrusek** [comment]
@fl4via PR and comment linked to newly created issue https://issues.jboss.org/browse/EJBCLIENT-302

**Pull request merged**